internal/services/datafactory/validate/ssh_private_key.go

```diff
@@ -0,0 +1,21 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func SSHPrivateKey(i interface{}, k string) (warnings []string, errors []error) {
+	value, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %q to be a string", k))
+		return
+	}
+	pattern := `(?s)-----BEGIN [A-Z0-9]+ PRIVATE KEY-----.*?-----END [A-Z0-9]+ PRIVATE KEY-----`
```

Comment:
I think this will need to be adjusted to also match header/footer that doesn't include the key type, e.g. `-----BEGIN PRIVATE KEY-----` / `-----END PRIVATE KEY-----`.